pan/legacy.pan

```diff
@@ -1,3 +1,12 @@
 declaration template pan/legacy;
 
 type legacy_binary_affirmation_string = string with match(SELF, "^(yes|no)$");
+
+function is_yes_no_true_false = {
+    v = ARGV[0];
+    if (is_boolean(v)) return(true);
+    if (is_string(v) && match(v, "^(yes|no)$")) return(true);
```

Comment:
as soon as we switch to this new type, we have to make changes to the components. but this is backwards compatible in pan, hence i think deprecation warnings are ok.